src/blocks/scratch3_data.js

```diff
@@ -0,0 +1,136 @@
+var Cast = require('../util/cast');
+
+function Scratch3DataBlocks(runtime) {
+    /**
+     * The runtime instantiating this block package.
+     * @type {Runtime}
+     */
+    this.runtime = runtime;
+}
+
+/**
+ * Retrieve the block primitives implemented by this package.
+ * @return {Object.<string, Function>} Mapping of opcode to Function.
+ */
+Scratch3DataBlocks.prototype.getPrimitives = function () {
+    return {
+        'data_variable': this.getVariable,
+        'data_setvariableto': this.setVariableTo,
+        'data_changevariableby': this.changeVariableBy,
+        'data_list': this.getListContents,
+        'data_addtolist': this.addToList,
+        'data_deleteoflist': this.deleteOfList,
+        'data_insertatlist': this.insertAtList,
+        'data_replaceitemoflist': this.replaceItemOfList,
+        'data_itemoflist': this.getItemOfList,
+        'data_lengthoflist': this.lengthOfList,
+        'data_listcontainsitem': this.listContainsItem
+    };
+};
+
+Scratch3DataBlocks.prototype.getVariable = function (args, util) {
+    var variable = util.target.lookupOrCreateVariable(args.VARIABLE);
+    return variable.value;
+};
+
+Scratch3DataBlocks.prototype.setVariableTo = function (args, util) {
+    var variable = util.target.lookupOrCreateVariable(args.VARIABLE);
+    variable.value = args.VALUE;
+};
+
+Scratch3DataBlocks.prototype.changeVariableBy = function (args, util) {
+    var variable = util.target.lookupOrCreateVariable(args.VARIABLE);
+    var castedValue = Cast.toNumber(variable.value);
+    var dValue = Cast.toNumber(args.VALUE);
+    variable.value = castedValue + dValue;
```

Comment:
Just noticed another casting weirdness here in 2.0:
https://github.com/LLK/scratch-flash/blob/master/src/interpreter/Interpreter.as#L715

Because it uses `Number` and not `numarg` (equivalent to our `Cast.toNumber`), this script:
<img width="154" alt="screen shot 2016-09-19 at 5 54 25 pm" src="https://cloud.githubusercontent.com/assets/120403/18650274/30914406-7e92-11e6-8437-e373a5b8e241.png">
Sets `lol` to NaN (not to 1). In the implementation I gave, I think `lol` would be 1.... should we reproduce the 2.0 way?